groups.go

```diff
@@ -90,6 +95,16 @@ type Group struct {
 
 	// Deprecated: Use EmailsEnabled instead
 	EmailsDisabled bool `json:"emails_disabled"`
+
+	// Deprecated: Use DefaultBranchProtectionDefaults instead
+	DefaultBranchProtection int `json:"default_branch_protection"`
```

Comment:
I don't think this is deprecated looking at the docs.

Reply:
@svanharmelen  - Hm, some of the conversations I've had with the GitLab employees certainly seem to insinuate it is, so I never thought about the fact that it isn't documented that way 🤔 

I'll talk with them on Monday and confirm it's status, then I'll either update the documentation or update this PR. Thanks for the great catch!

Reply:
https://gitlab.com/gitlab-org/gitlab/-/issues/462164#note_1998263497 for the question to GitLab